external/loaders/loaders/mappers/_hybrid.py

```diff
@@ -87,7 +88,7 @@ class FineResBudget(Protocol):
 
 
 def open_zarr(url, consolidated=False):
-    mapper = fsspec.get_mapper(url)
+    mapper = zarr.LRUStoreCache(fsspec.get_mapper(url), 128 * 2 ** 20)
```

Comment:
Is the idea with caching that MPI will call the same chunks multiple times since there's not chunking across tiles?

Reply:
The idea with caching is to take advantage of the fact that in prescribed runs our access pattern is to want subsequent steps in time (which are usually in the same chunk assuming the chunk size is greater than 1 in time). So the first time accessing data from a particular chunk, it gets loaded from GCS and then stored in the cache for subsequent timesteps.

w.r.t. the tiles... in the prog run, we separately load and scatter data from each tile using each tile's master rank. (for the case of 1 rank per tile, no scattering is actually required). So it's better to have a separate chunk for each tile, so we are not loading all the tiles on each rank.

Reply:
Thanks makes sense